mozetl/search/daily_search_rollups.py

```diff
@@ -1,309 +1,371 @@
-                row = [s.encode("utf-8") for s in row]
-                writer.writerow(row)
-            except Exception:
-                bad.append((k, v))
-    copy = "aws s3 cp {} {}".format(basename, ROLLUP_WRITE_STEM)
-    subprocess.check_call(copy, shell=True)
-    return bad
-
-
-def write_manifest(date, version, *paths):
-    """ Write a manifest file with the location of the daily rollup files. """
-    text = '\n'.join(paths) + '\n'
-    tries = version + 10
-    while True:
-        manifest_basename = 'daily-search-rollup-manifest-{}-v{}.txt'.format(
-            date, version)
-        path = MANIFEST_WRITE_STEM + manifest_basename
-        if subprocess.call("aws s3 ls {}".format(path), shell=True):
-            break
-        version += 1
-        if version == tries:
-            raise Exception("Can't find unwritten manifest at {}".format(path))
-    with open(manifest_basename, 'w') as f:
-        f.write(text)
-    copy = "aws s3 cp {} {}".format(manifest_basename, MANIFEST_WRITE_STEM)
-    subprocess.check_call(copy, shell=True)
-
-
-owner = 'spenrose'  # XXX read from environment
-owner += '@mozilla.com'
-
-
-def report(date, path, bad):
-    subject = "Daily search rollups for {}".format(date)
-    if bad:
-        subject += ": %d failures" % len(bad)
-    body = 'written to {}'.format(path)
-    if bad:
-        body += '\nproblem rows:\n'
-        for row in bad[:10]:
-            body += str(row) + '\n'
-
-    msg = Message()
-    msg.set_payload(body)
-    msg['Subject'] = subject
-    msg['To'] = owner
-    msg['From'] = owner
-    smtp = smtplib.SMTP('localhost')
-    smtp.sendmail(owner, owner, msg.as_string())
+
+    # persist the dataframe to disk
+    logging.info("Writing dataframe to {}/{}".format(bucket, key))
+    utils.write_csv_to_s3(dataframe.select(select_expr), bucket, key, header=False)
+
+    csv_paths = get_csv_locations(bucket, key)
+
+    # write the manifest to disk
+    write_manifest(bucket, prefix, mode, version, start_ds, csv_paths)
 
 
 @click.command()
-def main():
-    spark = (SparkSession
-             .builder
-             .appName("topline_dashboard")
-             .getOrCreate())
-
-    date = (DT.date.today() - DT.timedelta(1)).isoformat()
-    rows = generate_daily_search_rollup(spark, date)
-    dicty = coalesce(rows)
-
-    basename = BASENAME.format(date, WRITE_VERSION)
-    bad = dump_dict(dicty, basename)
-    path = get_s3_write_path(date)
-    try:
-        write_manifest(DT.date.today().isoformat(), 1, path)
-    except Exception:
-        report(date, path, ["Manifest writing failed"])
-    report(date, path, bad)
+@click.argument('start_date')
+@click.argument('mode', type=click.Choice(['daily', 'monthly']))
+@click.argument('bucket')
+@click.argument('prefix')
+@click.option('--input_bucket',
+              default='telemetry-parquet',
+              help='Bucket of the input dataset')
+@click.option('--input_prefix',
+              default='main_summary/v4',
```

Comment:
I think the proper way to address this is to have an up to date list of all logical tables leaving the low levels locations opaque to the user, like the hive metastore. 

I lean toward leaving this as is, since its consistent with other jobs that I have written. I think there is room for improvement here, but it should probably be more than a `MAIN_SUMMARY_PREFIX`/`MAIN_SUMMARY_VERSION` constant. I like the idea of having a centralized store with information about the location and version of all datasets, but it's also a lot of work to keep it up to date and robust.